### ISSUE TITLE
Should make-wrapper.sh reset the extraFlagsArray?

### ISSUE THREAD

**Issue by wmertens** [NixOS/nixpkgs#2537]
I noticed that make-wrapper.sh generates application wrappers that add extraFlagsArray flags to the invocation.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh#L75

Is this not a problem if extraFlagsArray gets set accidentally? Just wondering.

**Comment by Profpatsch** [comment]
Does the link still point to the right place?

**Comment by wmertens** [comment]
It moved to
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh#L81

On Tue, Sep 8, 2015, 23:28 Profpatsch <notifications@github.com> wrote:

> Does the link still point to the right place?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/issues/2537#issuecomment-138707570>.
>
-- 

Wout.
(typed on mobile, excuse terseness)

**Comment by vcunat** [comment]
Yeah, that would lead to unexpected behavior. Perhaps mark the variable local and clear it? (So the same-named variable wouldn't be shadowed when passing through exec.) I don't know bash well enough to be certain.

BTW, I can't find any relevant usage of the variable, so maybe the feature isn't really useful anyway...

**Comment by Profpatsch** [comment]
(triage) @wmertens Was this fixed in the meantime?

**Comment by wmertens** [comment]
No, it's still there, and in fact it doesn't seem to be used in nixpkgs: https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=extraFlagsArray (verified with grep -r as well)

The only use of the same name is [in builder.sh](https://github.com/NixOS/nixpkgs/blob/788800e437c4a0a25d95e217540bded68804b25e/pkgs/development/tools/misc/automake/builder.sh#L12-L28) which initializes it itself doesn't use make-wrapper.sh, and in simutrans where it seems to be a copy/paste.

So I propose we simply remove it from make-wrapper and simutrans. @vcunat @Profpatsch thoughts?

**Comment by wmertens** [comment]
I can't find a reference to it in nix and nixops either…

**Comment by Profpatsch** [comment]
The reasoning behind it is that you can programmatically change argument flags by invoking other programs in the wrapper (which is quite sensible):

```nix
with import <nixpkgs> {};

stdenv.mkDerivation {
  name = "test";
  src = ./.;

  buildInputs = [ makeWrapper ];

  installPhase = ''
    mkdir -p $out/bin
    echo -e '#!/bin/sh \necho "hello world, it’s a me, $@"' > $out/bin/hello
    chmod +x $out/bin/hello
    wrapProgram $out/bin/hello --run 'echo "WRAPPED!"' \
                               --run 'extraFlagsArray+="Mario!"'
  '';
}
```

I tested it, making it local seems to work well and should decrease the likelyhood of an accidental bug:
```
@@ -1,6 +1,6 @@
 let lib = import ./default.nix;
 
-inherit (builtins) trace attrNamesToStr isAttrs isFunction isList isInt
+inherit (builtins) trace deepSeq attrNamesToStr isAttrs isFunction isList isInt
         isString isBool head substring attrNames;
 
 inherit (lib) all id mapAttrsFlatten elem;

```

**Comment by wmertens** [comment]
@Profpatsch awesome, I did not expect that usage :) Definitely needs documentation, and in the mean time maybe a PR to make it a local and add a comment describing its purpose?

Let me know if you want me to do it instead.

**Comment by wmertens** [comment]
Fixed in #19328
